### ISSUE TITLE
Support pdf filename

### ISSUE THREAD

**Issue by kchousos** [mgmeyers/obsidian-zotero-integration#353]
Is there a way to get only the filename of the PDF file, without its full path?

**Comment by FeralFlora** [comment]
Yes, it's supported. It's under `{{attachments[0].title}}`. Where 0 is the first attachment. If you have more attachments, and want the titles of all of them, you can loop over the attachments array and then access the of each attachment in this loop.

You can see all this data in Zotero Integration's Data explorer, found using the "Data explorer" command. You can also right-click elements in the data explorer to copy the template path to that element.

**Comment by aetourneau** [comment]
> Yes, it's supported. It's under `{{attachments[0].title}}`. Where 0 is the first attachment. If you have more attachments, and want the titles of all of them, you can loop over the attachments array and then access the of each attachment in this loop.
> 
> You can see all this data in Zotero Integration's Data explorer, found using the "Data explorer" command. You can also right-click elements in the data explorer to copy the template path to that element.

Actually, `{{attachments[0].title}}` seems to take the name of the PDF in the Zotero database and not the actual filename:

Zotero 7 renames PDFs attached to metadata as things like "Full Text PDF" while the filename is of the format "{{author}} - {{date}} - {{title}}.pdf": [https://www.zotero.org/support/kb/attachment_title_vs_filename](https://www.zotero.org/support/kb/attachment_title_vs_filename)

So using `{{attachments[0].title}}` links to a generic name shared by many PDFs but not the actual file. Using `{{attachments[0}.path}` is an alternative but not robust if you sync your vault on several machines. It's not useful either if you keep the PDFs inside your vault and you want to refer to them, because you need neither the absolute path nor the value provided with `{{attachments[0].title}}`.

Filename metadata seems accessible in Zotero. When you look in the "Attachment Info" tab, you see info such as "Title" (what `{{attachments[0].title}}` returns) and "Filename". But this filename metadata does not seem to be read by Zotero Integration plugin as I can't find it in the data explorer.
